test/integration/webhook_test.go

```diff
@@ -684,12 +680,10 @@ func waitForWebhookServiceConnective(ctx context.Context, configResourceName str
 	return networking.WaitForConnectionOnServicePort(waitCtx, env.Cluster().Client(), consts.ControllerNamespace, svcName, svcPort, 10*time.Second)
 }
 
-func ensureAdmissionRegistration(ctx context.Context, configResourceName string, rules []admregv1.RuleWithOperations) (func() error, error) {
+func ensureAdmissionRegistration(ctx context.Context, t *testing.T, configResourceName string, rules []admregv1.RuleWithOperations) {
 	svcName := fmt.Sprintf("webhook-%s", configResourceName)
 	svcCloser, err := ensureWebhookService(ctx, svcName)
```

Comment:
We can make an action item or issue to refactor all those `ensure...()` calls to use `t.Cleanup`

Reply:
I'm doing PR for this (almost as fast as writing an issue), I will push it after merging this PR to avoid confilicts